File: app/src/main/java/com/zxcx/zhizhe/ui/my/followUser/UnFollowConfirmDialog.java
package com.zxcx.zhizhe.ui.my.followUser;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;
import com.zxcx.zhizhe.R;
import com.zxcx.zhizhe.event.UnFollowConfirmEvent;
import com.zxcx.zhizhe.mvpBase.CommonDialog;
import org.greenrobot.eventbus.EventBus;

/**
 * Created by anm on 2017/7/21.
 */

public class UnFollowConfirmDialog extends CommonDialog {
	
	Unbinder unbinder;
	@BindView(R.id.tv_dialog_logout_title)
	TextView tvDialogLogoutTitle;
	@BindView(R.id.tv_dialog_cancel)
	TextView tvDialogCancel;
	@BindView(R.id.tv_dialog_confirm)
	TextView tvDialogConfirm;
	
	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
		Bundle savedInstanceState) {
		getDialog().requestWindowFeature(Window.FEATURE_NO_TITLE);
		View view = inflater.inflate(R.layout.dialog_logout, container);
		unbinder = ButterKnife.bind(this, view);
		return view;
	}
	
	@Override
	public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);
		tvDialogLogoutTitle.setText("是否取消对该作者的关注");
	}
	
	@Override
	public void onDestroyView() {
		super.onDestroyView();
		unbinder.unbind();
	}
	
	@OnClick(R.id.tv_dialog_cancel)
	public void onMTvDialogCancelClicked() {
		this.dismiss();
	}
	
	@OnClick(R.id.tv_dialog_confirm)
	public void onMTvDialogConfirmClicked() {
		EventBus.getDefault().post(new UnFollowConfirmEvent(getArguments().getInt("userId")));
		this.dismiss();
	}
}

File: app/src/main/java/com/zxcx/zhizhe/ui/my/followUser/UnFollowConfirmDialog.kt
package com.zxcx.zhizhe.ui.my.followUser

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import com.zxcx.zhizhe.R
import com.zxcx.zhizhe.event.UnFollowConfirmEvent
import com.zxcx.zhizhe.mvpBase.CommonDialog
import kotlinx.android.synthetic.main.dialog_single.*
import org.greenrobot.eventbus.EventBus

/**
 * Created by anm on 2017/7/21.
 * 取消关注弹窗
 */

class UnFollowConfirmDialog : CommonDialog() {

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
	                          savedInstanceState: Bundle?): View? {
		dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
		val view = inflater.inflate(R.layout.dialog_single, container)
		return view
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)
		tv_dialog_title.text = "是否取消对该用户的关注"
	}

	override fun setListener() {
		super.setListener()
		tv_dialog_cancel.setOnClickListener {
			this.dismiss()
		}
		tv_dialog_confirm.setOnClickListener {
			EventBus.getDefault().post(arguments?.getInt("userId")?.let { it1 -> UnFollowConfirmEvent(it1) })
			this.dismiss()
		}
	}
}
